### PR TITLE
[WIP] Add nette/coding-standard

### PR DESCRIPTION
**Pull request by TomasVotruba**
Based on discussion with @dg , this the best repo to start with.

## Goals

- identify useful checkers
- drop the rest
- see, how it works as dependency package

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Does anyone have idea why dependencies are not installed? 

Probably many dev dependencies.

This waits on https://github.com/squizlabs/PHP_CodeSniffer/pull/1442#issuecomment-302253910 then

**Comment by dg** [comment]
It must be installed the same way as nette codechecker, adding as composer dependency is bad way.

**Comment by TomasVotruba** [comment]
@dg Could you elaborate, why do you think so?

If somebody will make a PR, how could he fix the code locally then?

I'll look on this after CodeSniffer 3.0.1 is released.

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,9 @@
 		"nette/security": "^3.0",
 		"latte/latte": "^3.0",
 		"tracy/tracy": "^2.4",
-		"mockery/mockery": "^1.0"
+		"mockery/mockery": "^1.0",
+		"nette/coding-standard": "@dev",
+		"symplify/easy-coding-standard": "@dev"
```
Comment (f3l1x) [inline composer.json]: `symplify/easy-coding-standard` is also needed? I thought it's in nette/coding-standard repo, isn't it?

Comment (TomasVotruba) [inline composer.json]: Have you tried installing "nette/coding-standard" as dependency? I think it should fail without this.

**Comment by dg** [comment]
The developer just installs nette/coding-standard locally and runs it.

**Comment by TomasVotruba** [comment]
@dg What is the advantage of this?

**Comment by dg** [comment]
This is way how to install and run easy-coding-standard (but it currently fails on autoloading error).
```
composer create-project symplify/easy-coding-standard --no-dev
easy-coding-standard\bin\easy-coding-standard check src
```

**Comment by dg** [comment]
It is not about advantage, it is the only possible solution how to install it. Because easy-coding-standard is separate tool, it is not part of nette/application. It also uses nette/application, so there is no way how to install nette/application into nette/application.

**Comment by TomasVotruba** [comment]
I see, thanks for explaining.

I have one last blind spot. If I install it locally to the package like this, how do I make sure I won't accidentally commit it?

**Comment by dg** [comment]
> If I install it locally to the package like this, how do I make sure I won't accidentally commit it?

Of course, it's better to install it in another directory. I can develop more nette libraries and it makes no sense to install easy-coding-standard multiple times. But if I want to have it in library folder, I'd add a line to `.git/info/exclude`.

>  I wonder what is application actually needed for. Probably not.

It's not just about `application`, you should not use any library from nette.

PhpStan is similarly broken, so it cannot be used for Nette or Symfony.

**Comment by TomasVotruba** [comment]
I see. I wish there was more elegant solution than installing a project and ignoring it.

**Comment by dg** [comment]
Possible workaround is to install it into `vendor`, because it is ignored by all libraries `composer create-project symplify/easy-coding-standard vendor/easy-coding-standard --no-dev`

But I wouldn't want have it in the project at all, because I it should be hidden for PhpStorm too etc.

**Comment by TomasVotruba** [comment]
This can be closed, right?

**Pull request closed without merging**